### ISSUE TITLE
Re-evaluate how to check for commit-ish in commit/tag/branch of git modules

### ISSUE THREAD

**Issue by barthalion** [flathub/flatpak-builder-lint#16]
flatpak-builder allows arbitrary commit-ish in all three commit/tag/branch keys. We need to check at least the length of the value to ensure it's actually a commit, and someone doesn't bypass the check by putting tag/branch name in `commit`.

**Comment by hfiguiere** [comment]
or "branch" as "tag"

**Comment by prodrigestivill** [comment]
It should be allowed to use a branch in sources?
Because, it is currently not allowing it.

The actual lint check doesn't allow to have sources that checkout a commit from a branch.
The org.flathub.flatpak-external-data-checker indeed updates to the latest commit from the branch with the following example, but the lint fails.

```yaml
      - type: git
        url: https://github.com/xxx/xxx-translations.git
        branch: master
        commit: 793ae9326330986ab9adfcc380be1561d09365a1
        dest: src/translations
```

**Comment by hfiguiere** [comment]
no it shouldn't. A branch is a moving target. We want reproducable build.

And here, flatpak-builder will fail if commit doesn't match branch. For a good reason.

**Comment by prodrigestivill** [comment]
I thought flatpak-builder would prioritize checking out by commit if both (tag|branch) and commit  are present.
And (branch|tag) was only used (if they are present together with the commit) to instruct flatpak-external-data-checker to update the commit entry.

In any case, thanks for the clarification.

**Comment by barthalion** [comment]
We definitely want to keep branch+commit combination working, and I've made an apparently poor attempt at this at https://github.com/flathub/flatpak-builder-lint/blob/master/flatpak_builder_lint/checks/modules.py#L22.

**Comment by hfiguiere** [comment]
> We definitely want to keep branch+commit combination working, and I've made an apparently poor attempt at this at https://github.com/flathub/flatpak-builder-lint/blob/master/flatpak_builder_lint/checks/modules.py#L22.

Not sure what you mean but if the commit specified is not the head of the specifc branch, then flatpak-builder will bail out. Changing this would require fixing flatpak-builder.

As for having this work for f-e-d-c, we discussed this on matrix and a few of us believe it should be a parameter to the git checker
